src/tests/stdlib.rs

```diff
@@ -0,0 +1,396 @@
+    let temp_dir = tempdir().expect("Failed to create temporary directory");
+    let file_path = temp_dir.path().join("test_file.txt");
+
+    let mut file = fs::File::create(&file_path).expect("Failed to create temporary file");
+    file.write_all(b"This is sample file.").expect("Failed to write to temporary file");
+
+    let code = format!(
+        "
+        import * from \"std\"
+        main {{
+            let f = file_read(\"{file_path}\") failed {{ echo \"Failed\" }}
+            echo f
+        }}
+        ",
+        file_path = file_path.to_str().unwrap()
+    );
+
+    test_amber!(code, "This is sample file.");
+
+    fs::remove_file(&file_path).expect("Failed to delete temporary file");
+}
+
+#[test]
+fn file_write() {
+    let temp_dir = tempdir().expect("Failed to create temporary directory");
+    let file_path = temp_dir.path().join("test_file.txt");
+
+    let code = format!(
+        "
+        import * from \"std\"
+        main {{
+            unsafe file_write(\"{file_path}\", \"Hello, Amber!\")
+        }}
+        ",
+        file_path = file_path.to_str().unwrap()
+    );
+
+    test_amber!(code, "");
+
+    let mut file_content = String::new();
+    fs::File::open(&file_path)
+        .expect("Failed to open temporary file")
+        .read_to_string(&mut file_content)
+        .expect("Failed to read from temporary file");
+
+    assert_eq!(file_content.trim(), "Hello, Amber!");
+
+    fs::remove_file(&file_path).expect("Failed to delete temporary file");
+}
+
+#[test]
+fn file_append() {
+    let temp_dir = tempdir().expect("Failed to create temporary directory");
+    let file_path = temp_dir.path().join("test_file.txt");
+
+    let mut initial_file = fs::File::create(&file_path).expect("Failed to create temporary file");
+    initial_file.write_all(b"Hello, Amber!\n").expect("Failed to write to temporary file");
+
+    let code = format!(
+        "
+        import * from \"std\"
+        main {{
+            unsafe file_append(\"{file_path}\", \"Appending this line.\")
+        }}
+        ",
+        file_path = file_path.to_str().unwrap()
+    );
+
+    test_amber!(code, "");
+
+    let mut file_content = String::new();
+    fs::File::open(&file_path)
+        .expect("Failed to open temporary file")
+        .read_to_string(&mut file_content)
+        .expect("Failed to read from temporary file");
+
+    assert_eq!(file_content.trim(), "Hello, Amber!\nAppending this line.");
+
+    fs::remove_file(&file_path).expect("Failed to delete temporary file");
+}
+
+#[test]
+fn split() {
+    let code = "
+        import * from \"std\"
+        main {
+            let array = split(\"apple,banana,cherry\", \",\")
+            echo array[1]
+        }
+    ";
+    test_amber!(code, "banana")
+}
+
+#[test]
+fn join() {
+    let code = "
+        import * from \"std\"
+        main {
+            echo join([\"apple\", \"banana\", \"cherry\"], \", \")
+        }
+    ";
+    test_amber!(code, "apple,banana,cherry")
```

Comment:
```suggestion
    test_amber!(code, "apple, banana, cherry")
```

Since we join with the comma and space between - why don't we add the space there as well?

Reply:
Even if multiple characters are specified for `IFS`, it will only be one character when output, so for this case, I'll set it to just `,`.